source/opt/pass.cpp

```diff
@@ -60,6 +60,46 @@ bool Pass::ProcessEntryPointCallTree(
   return modified;
 }
 
+bool Pass::ProcessReachableCallTree(
+    ProcessFunction& pfn, ir::Module* module) {
+  // Map from function's result id to function
+  std::unordered_map<uint32_t, ir::Function*> id2function;
+  for (auto& fn : *module)
+    id2function[fn.result_id()] = &fn;
+  // Process call tree
+  bool modified = false;
+  std::queue<uint32_t> todo;
+  std::unordered_set<uint32_t> done;
+
+  // Add all entry points since they can be reached from outside the module.
+  for (auto& e : module->entry_points())
+    todo.push(e.GetSingleWordInOperand(kEntryPointFunctionIdInIdx));
+
+  // Add all exported functions since they can be reached from outside the
+  // module.
+  for (auto& a : module->annotations()) {
+    if (a.opcode() == SpvOp::SpvOpDecorate) {
+      if (a.GetSingleWordOperand(1) == SpvDecoration::SpvDecorationLinkageAttributes) {
+        uint32_t lastOperand = a.NumOperands()-1;
+        if (a.GetSingleWordInOperand(lastOperand) == SpvLinkageType::SpvLinkageTypeExport) {
```

Comment:
In this instruction GetSingleWordOperand is the same as GetSingleWordInOperand.  So I suggest using one or the other but not both.